File: 26.cpp
//
// Created by DuLi on 2023/9/3.
//

#include <iostream>
using std::vector;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int p1 = 0;

        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] != nums[p1]){
                p1++;
                nums[p1] = nums[i];
            }
        }
        return p1 + 1;
    }
};

File: 26.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        p1 = 0
        k = 0
        j = 100000

        for i in range(len(nums)):
            if nums[i] != j:
                j = nums[i]
                nums[p1] = nums[i]
                k += 1
                p1 += 1
            else:
                continue
        return k

nums = [0,0,1,1,1,2,2,3,3,4]
sol = Solution()
result = sol.removeDuplicates(nums)
print(result)


